### ISSUE TITLE
[Chore] Incomplete test coverage

### ISSUE THREAD

**Issue by longnd** [aungkoko1234/web-scrapper-backend#23]
Issue
---
While some tests were created for the controller and services, but they're mainly the setup part or non-essential APIs. The core business logic of the domains in KeyWord service and KeyWordController are not tested: the file upload and the job to process the keyword search & scraping the result.
The goal is not to always get 100 test coverage (it's a nice and motivating goal for a team) and I also value the opinion of focusing on critical tests first, but in this case, the test coverage is not sufficient. Also not covering the services is not satisfying in the way that these parts are the hardest to test and you could have demonstrated your unit testing skills better.

**Comment by aungkoko1234** [comment]
Yeah.. it is difficult to mock the queue and can have enough time to focus to test. I should have written the unit testing better than that.
